frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/print/elements/impl/EdgePrintElement.java

```diff
@@ -0,0 +1,55 @@
+
+package com.virtuslab.gitmachete.frontend.graph.print.elements.impl;
+
+import lombok.Getter;
+import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.graph.api.GraphEdge;
+import com.virtuslab.gitmachete.frontend.graph.coloring.PrintElementColorManager;
+import com.virtuslab.gitmachete.frontend.graph.print.elements.api.IEdgePrintElement;
+
+@Getter
+public class EdgePrintElement extends PrintElementWithGraphElement implements IEdgePrintElement {
+
+  private final Type type;
+
+  public EdgePrintElement(@NonNegative int rowIndex,
+      @NonNegative int positionInRow,
+      Type type,
+      GraphEdge graphEdge,
+      PrintElementColorManager printElementColorManager) {
+    super(rowIndex, positionInRow, graphEdge, printElementColorManager);
+    this.type = type;
+  }
+
+  @Override
+  public boolean equals(@Nullable Object o) {
```

Comment:
Ditto, add `equals` spec to base class

Reply:
Actually the depending on the default implementation is enough